src/main/scala/org/allenai/plugins/DeployPlugin.scala

```diff
@@ -391,34 +595,12 @@ object DeployPlugin extends AutoPlugin {
 
   /* ==========>  The main deploy task.  <========== */
 
-  /** Parser used for reading deploy-config overrides into a [[Config]] object.
-    * Overrides are passed using property-style syntax: -Dpath1=value1 -Dpath2=value2, etc.
-    */
-  lazy val overridesParser: State => Parser[Config] = { _ =>
-    // Parser for a single property-style config override.
-    // Equivalent of extracting from the regex: `-D([^=])*=(.*)`.
-    val overrideParser: Parser[(String, String)] = token(
-      ("-D" ~> charClass(_ != '=').*.string) ~ ('=' ~> StringBasic),
-      "-D<deploy.key>=<override>"
-    )
-
-    // Parser for one or more space-delimited config overrides.
-    val overridesParser: Parser[Seq[(String, String)]] =
-      rep1sep(overrideParser, Space)
-
-    // Parser which folds some number of overrides into a Config object.
-    overridesParser map { overrides =>
-      overrides.foldLeft(ConfigFactory.empty) {
-        case (config, (path, value)) => config.withValue(path, ConfigValueFactory.fromAnyRef(value))
-      }
-    }
-  }
-
-  /** Parser used for reading input for the deploy task. */
-  lazy val deployParser: State => Parser[(Config, String)] = { state =>
-    Space ~> ((overridesParser(state) <~ Space).? ~ deployTargetParser(state)) map {
-      case (overrides, env) => (overrides getOrElse ConfigFactory.empty, env)
-    }
+  /** Wrapper for information used to access a remote host. */
+  class SshInfo(val user: String, val host: String, keyFile: Option[String]) {
+    val keyfileFlag = keyFile map { Seq("-i", _) } getOrElse Seq()
```

Comment:
You can also call `toIterable` on `keyFile` before calling `map`, instead of using `getOrElse`.